src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -1316,6 +1317,40 @@ Value ByteCodeInterpreter::interpret(ExecutionState& state, ByteCodeBlock* byteC
                 NEXT_INSTRUCTION();
             }
 
+            DEFINE_OPCODE(GeneratorComplete)
+                :
+            {
+                GeneratorComplete* code = (GeneratorComplete*)programCounter;
+                ec->generatorTarget()->asGeneratorObject()->setState((GeneratorState)(GeneratorState::CompletedReturn + !!code->m_isThrow));
+                ADD_PROGRAM_COUNTER(GeneratorComplete);
+                NEXT_INSTRUCTION();
+            }
+
+            DEFINE_OPCODE(Yield)
+                :
+            {
+                Yield* code = (Yield*)programCounter;
+                GeneratorObject* gen = ec->generatorTarget()->asGeneratorObject();
+
+                gen->setState(GeneratorState::SuspendedYield);
+                ADD_PROGRAM_COUNTER(Yield);
+                gen->setByteCodePosition((char*)programCounter - codeBuffer);
+                gen->setResumeValueIdx(code->m_dstIdx);
+                return code->m_yieldIdx == std::numeric_limits<ByteCodeRegisterIndex>::max() ? Value() : registerFile[code->m_yieldIdx];
```

Comment:
`std::numeric_limits<ByteCodeRegisterIndex>::max()` can be replaced with `REGISTER_LIMIT`